content/artifacts/highlight-active-artifact-planets/plugin.js

```diff
@@ -0,0 +1,173 @@
+// Display the planet with active artifact, specially for cannon.
+
+import { 
+  BiomeNames
+} from "https://cdn.skypack.dev/@darkforest_eth/types";
+
+import {
+  coords
+} from 'https://plugins.zkga.me/utils/utils.js';
+
+import {
+  ArtifactType,
+  ArtifactTypeNames
+} from "https://cdn.skypack.dev/@darkforest_eth/types";
+
+const ARTIFACT_TYPES = Object.values(ArtifactType).filter((type) => type !== ArtifactType.Unknown).map((type) => ({ value: type, text: ArtifactTypeNames[type] }));
+
+
+function getPlayerColor(ethAddress) {
+  return df.getProcgenUtils().getPlayerColor(ethAddress);
+}
+
+
+function drawArc(ctx, planet, color, artifactTypes){
+  if(!planet) {
+    return false;
+  }
+
+  let artifact = df.getActiveArtifact(planet);
+  if (!artifact) {
+    return false;
+  }
+
+  if (artifactTypes.indexOf(artifact.artifactType ) == -1)  {
+    return false;
+  }
+
+  let percent = 1;
+  if (artifact.artifactType == 7) {
+    let timePass = Math.floor(Date.now() / 1000 - artifact.lastActivated);
+    percent = timePass / ( 4 * 60 * 60);
+  }
+
+  const viewport = ui.getViewport();
+  ctx.strokeStyle = color;
+  ctx.lineWidth = 5;
+  if (!planet.location) {
+    return false;
+  }
+  const {x, y} = viewport.worldToCanvasCoords(planet.location.coords);
+  const range = planet.range * 0.01 * 20;
+  const trueRange = viewport.worldToCanvasDist(range);
+  ctx.beginPath();
+  ctx.arc(x, y, trueRange, 0, 2 * Math.PI * percent);
+  ctx.stroke();
+  if (artifact.artifactType == 7 && percent < 0.99) {
+    ctx.lineWidth = 1;
+    ctx.beginPath();
+    ctx.arc(x, y, trueRange, 0, 2 * Math.PI);
+    ctx.stroke();
+  }
+  return true;
+}
+
+
+class Plugin {
+  constructor() {
+    this.message = document.createElement('div');
+    this.showGroup = 'My';
+    this.showArifactTypes = [7];
+  }
+
+
+  async render(container) {
+    container.style.minHeight = 'unset';
+    container.style.width = '300px';
+
+    let groupNameLabel = document.createElement('div');
+    groupNameLabel.innerHTML = 'Highlight all active planet or mine?'
```

Comment:
Maybe this text could be changed to
'Highlight all active artifacts or just mine?'

Reply:
OK, thanks